src/common/snippets/docs/snippets_design_guide.md

```diff
@@ -1,301 +1,465 @@
+The described fusing technique is well-established and widely used across the industry. It is especially profitable, since it allows one to reduce execution time without any associated accuracy penalties.
+Note however, that fusings are not entirely free, since they are introducing additional complications to existing pipeline: 
+* First, you have to implement single `Op` jit kernels (`Load`, `Store`, `Op instructions` blocks).
+* Next, you need a mechanism that allows to merge several jit kernels into one (execute several `Op instructions` blocks inside one kernel).
+* Finally, you need to develop an algorithm that understands what kinds of `Ops` can be fused, and accounts for their data dependencies.
 
-##### Dialect conversion
+A common approach to tackle these issues is to create a set of hand-crafted fusing transformations. For instance, OneDNN offers a post-ops fusing mechanism, and it is relatively easy to implement fusigns of simple elementwise operations using this mechanism. This approach has proven to be both convenient and effective in the past, however it does not meet the demands of the present. Indeed, a rapidly increasing variety of deep-learning models creates an even faster growing demand for new fusing patters. This makes it increasingly expensive to support and extend the set of hand-crafted fusing transformations, so a new more scalable approach is required. The `Snippets` were specifically designed to address this issue.
 
-The goal for this step is to transform a subgraph (body function) into a form possible for code generation. Input for this step is a subgraph in a canonical form. Output is a subgraph in snippets dialect.
+The `Snippets` is a code generation framework that provides a scalable approach to operations. fusings and  enablement. One of the most distinctive feature of `Snippets` is a flexible yet universal design. They are not bound to a specific backend or concrete kernel implementations, but provide a jit kernel interface instead. This interface allows `Snippets` to perform register assignment as well as a wide range of other backend-independent optimizations. `Snippets` are also highly configurable and allow a backend to provide a custom optimizations that can be executed at different stages of the pipeline. This unique combination of flexibility and scalability is driven by `Snippets` design, so let's discuss it in more detail. 
```

Comment:
Seems like there is a mark
`The Snippets is a code generation framework that provides a scalable approach to operation fusings and enablement.`

Reply:
right, thanks